source/guides/1.1/repositories/overview.md

```diff
@@ -218,6 +217,72 @@ operator = OperatorRepository.new.create(name: "Jane")
   # => #<Operator:0x007f8e43cbcea0 @attributes={:id=>1, :name=>"Jane"}>
 ```
 
+There's a few caveats tho (as always with legacy databases):
+
+Note that all attributes need to be mapped or you will get an error when booting up the application that will look like this:
+
+```ruby
+class OperatorRepository < Hanami::Repository
+  self.relation = :t_operator
+  
+  mapping do
+    attribute :id, from: :operator_id
+  end
+end
+
+  # => Hanami::Model::Error: key not found: :s_name
+```
+That happens because we infer the schema for the table and mapping only really changes the `Hanami::Entity` and commands that operate over it (like `create` and `update`).
+
+A query on the other hand will still need to use the original attribute name:
+
+```ruby
+class OperatorRepository < Hanami::Repository
+  self.relation = :t_operator
+  
+  mapping do
+    attribute :id,   from: :operator_id
+    attribute :name, from: :s_name
+  end
+  
+  def by_name(name)
+    t_operator.where(s_name: name)
+  end
+end
+```
+
+If you want to use the more semantic names (or with just a subset of attributes) you'll need to define a schema and a mapping:
+
+```ruby
+class OperatorRepository < Hanami::Repository
+  self.relation = :t_operator
+  
+  schema do
+    attribute :operator_id, Hanami::Model::Sql::Types::Int.meta(primary_key: true, alias: :id)
```

Comment:
just idea, but maybe we can use more friendly types here?

Reply:
While we probably could use `Dry::Types` not sure if it wouldn't lead to confusion as the end user might think something along the likes of "oh I need yet another lib to do this".